common/beerocks/bcl/include/bcl/beerocks_promise.h

```diff
@@ -23,38 +24,30 @@ template <class T> class promise {
     // Set value for waiting threads
     void set_value(const T value)
     {
-        int err;
-        if ((err = pthread_mutex_lock(&m_mut)) != 0) {
-            LOG(ERROR) << "pthread_mutex_lock failed, error code: " << err;
-            return;
-        }
+        int err = pthread_mutex_lock(&m_mut);
+        assert(err == 0);
         m_value  = value;
         m_signal = 1;
-        if ((err = pthread_cond_broadcast(&m_cond)) != 0)
-            LOG(ERROR) << "pthread_cond_broadcast failed, error code: " << err;
-        if ((err = pthread_mutex_unlock(&m_mut)) != 0)
-            LOG(ERROR) << "pthread_mutex_unlock failed, error code: " << err;
+        err      = pthread_cond_broadcast(&m_cond);
+        assert(err == 0);
```

Comment:
Please use mapf_assert from mapf/common/err.h.

It does something different than assert at the moment, but with a wrapper it's easier to change to standard assert (and use LOG for logging).

(nitpick)

Reply:
Actually, not such a nitpick, as you can see in https://gitlab.com/prpl-foundation/prplMesh/-/jobs/411168392

With -DNDEBUG (i.e., when building for release), `assert` is defined as empty, so gcc thinks err is not used.